packages/ui/src/components/Designer/RightSidebar/DetailView/index.tsx

```diff
@@ -75,22 +76,39 @@ const DetailView = (
   }, [form, activeSchema]);
 
   const handleWatch = (newSchema: any) => {
-    const changes = [];
+    let changes: ChangeSchemaItem[] = [];
     for (let key in newSchema) {
       if (['id', 'content'].includes(key)) continue;
+
+      // [position] Return the flattened position to its original form.
+      if (key === 'x') key = 'position.x';
+      if (key === 'y') key = 'position.y';
```

Comment:
we need to flatten before we check if the value has changed, previously this was comparing against undefined so we always considered `x` and `y` had changed.